### ISSUE TITLE
Tracing decorators could allow more customization of Span.

### ISSUE THREAD

**Issue by anuraaga** [line/armeria#1332]
I think right now, it's not easy to customize a span. While thinking about https://github.com/openzipkin/zipkin/issues/2167 I realized that being able to set `Span.remoteServiceName` based on the host would be convenient. I think we could add a customizer given the `ctx` and `Span`.

Of course, a delegate decorator could do this already but it's pretty cumbersome.

**Comment by adriancole** [comment]
agreed.. we have some practice in brave both for normal http and grpc. I think it is called Parser when we are scraping something for tags
